pkg/k8s/scanner/io.go

```diff
@@ -13,31 +14,58 @@ import (
 	"github.com/aquasecurity/trivy/pkg/log"
 )
 
-var r = regexp.MustCompile("\\\\|/|:|\\*|\\?|<|>")
+var r = regexp.MustCompile("[\\\\/:*?<>]")
 
-func createTempFile(artifact *artifacts.Artifact) (string, error) {
+func generateTempFileByArtifact(artifact *artifacts.Artifact, tempFolder string) (string, error) {
 	filename := fmt.Sprintf("%s-%s-%s-*.yaml", artifact.Namespace, artifact.Kind, artifact.Name)
-
 	if runtime.GOOS == "windows" {
 		// removes characters not permitted in file/directory names on Windows
 		filename = filenameWindowsFriendly(filename)
 	}
-	file, err := os.CreateTemp("", filename)
+	file, err := os.CreateTemp(tempFolder, filename)
 	if err != nil {
-		return "", xerrors.Errorf("creating tmp file error: %w", err)
+		return "", xerrors.Errorf("failed to create temporary file: %w", err)
 	}
+	shouldRemove := false
 	defer func() {
 		if err := file.Close(); err != nil {
-			log.Error("Failed to close temp file", log.String("path", file.Name()), log.Err(err))
+			log.Error("Failed to close temp file", log.FilePath(file.Name()), log.Err(err))
+		}
+		if shouldRemove {
+			removeFile(file.Name())
 		}
 	}()
-
 	if err := yaml.NewEncoder(file).Encode(artifact.RawResource); err != nil {
-		removeFile(filename)
-		return "", xerrors.Errorf("marshaling resource error: %w", err)
+		shouldRemove = true
+		return "", xerrors.Errorf("failed to encode artifact: %w", err)
+	}
+	return filepath.Base(file.Name()), nil
+}
+
+// generateTempFolder creates a folder with yaml files generated from kubernetes artifacts
+// returns a folder name, a map for mapping a temp target file to k8s artifact and error
+func generateTempFolder(arts []*artifacts.Artifact) (string, map[string]*artifacts.Artifact, error) {
```

Comment:
nit: I think `directory` is more common than `folder` in UNIX. This function actually calls `MkdirTemp`, not `MkfolderTemp`.

Reply:
rename https://github.com/aquasecurity/trivy/pull/7690/commits/84cacc5dc531c09ccce443271b6f0f37bcf05b58